docs/messaging/messaging-spans.md

```diff
@@ -271,6 +272,11 @@ batch of messages, or for no message at all (if it is signalled that no
 messages were received). For each message it accounts for, the "Deliver" or
 "Receive" span SHOULD link to the message's creation context.
 
+"Process" spans MAY be created in addition to "Deliver" and "Receive" spans,
+for operations of processing a single message or a batch of messages. For each
```

Comment:
It's hard to distinguish deliver and process:

- For example, Azure messaging client libraries have processing client (e.g. [EventHubProcessingClient](https://learn.microsoft.com/java/api/com.azure.messaging.eventhubs.eventprocessorclient?view=azure-java-stable) with API like `processEvents`  callback. 
- Azure also supports HTTP-push like [EventGrid](https://learn.microsoft.com/en-us/azure/event-grid/push-delivery-overview) along with the extension supporting Azure Functions triggers. 
- Such processor clients are designed is that if processing fails, messages should be redelivered. I.e. settlement is expected to happen within processor callback (but it's effectively up to the user in common case).

Similarly, [AWS SNS](https://docs.aws.amazon.com/sns/latest/dg/sns-message-delivery-retries.html#creating-delivery-policy) and [GCP PubSub](https://cloud.google.com/pubsub/docs/push) supports HTTP-push with redeliveries if request fails.

The typical [RabbitMQ](https://www.rabbitmq.com/confirms.html#acknowledgement-modes) use-case is also to settle message within consume/deliver callback.

So, deliver operation (at least for the common case)  is expected to also settle a message within it's scope and messaging brokers are designed to retry delivery if delivery fails or is negatively acked.

Also, deliver operations are those which can be automatically instrumented (by SDKs, FaaS extensions, etc).

Given all of this, I find it hard to draw a line between processing and delivery.

So, I think we should either:
- merge them together and maybe allow using different terminology
- clearly explain the difference  (e.g. processing is an application-level span which traces individual messages within a batch. We don't expect it to be created automatically, but we recommend (lowercase) everyone creating it to use the following semantics ...)

Reply:
I think the case where we would have a clear distinction is the pull-based scenario, like you mentioned:

> or a span that traces processing when pull-based receive is used - in this case it's semantics SHOULD match deliver operation. Sometimes it can be created automatically (e.g. by integration libraries such as [spring-kafka](https://www.baeldung.com/spring-kafka#2-consuming-messages))

So I think then it makes sense to have the process semantics, but make it clear that case.

Reply:
Discussed in the workgroup meeting: we will change this PR to merge "process" and "deliver" spans and metrics.

Reply:
This PR now unifies "deliver" and "process" operations (and metrics).

@open-telemetry/semconv-messaging-approvers, please have a look.